runner/runner_test.go

```diff
@@ -250,15 +253,20 @@ func Test_runTracerouteMulti(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			counter = 1
+			counter.Store(0)
 			runTracerouteOnceFn = tt.tracerouteOnceFn
 			defer func() { runTracerouteOnceFn = runTracerouteOnce }()
 
 			results, err := runTracerouteMulti(context.Background(), tt.params, 42)
 			if tt.expectedError != "" {
 				assert.EqualError(t, err, tt.expectedError)
 			}
-			require.Equal(t, tt.expectedResults, results)
+			expectedResultsJson, err := json.MarshalIndent(tt.expectedResults, "", "  ")
```

Comment:
```suggestion
			// Sort results by destination IP for deterministic comparison
			sort.Slice(results.Traceroute.Runs, func(i, j int) bool {
			  return bytes.Compare(results.Traceroute.Runs[i].Destination.IPAddress,
			  results.Traceroute.Runs[j].Destination.IPAddress) < 0
			})
			expectedResultsJson, err := json.MarshalIndent(tt.expectedResults, "", "  ")
```

The test is still flaky - the order of t he results is indeterminate.  I was doing something like this to sort the results before comparing before I saw your PR.